dandiapi/api/views/dandiset.py

```diff
@@ -485,7 +485,10 @@ def unembargo(self, request, dandiset__pk):
     # TODO: move these into a viewset
     @action(methods=['GET', 'PUT'], detail=True)
     def users(self, request, dandiset__pk):  # noqa: C901
-        dandiset: Dandiset = self.get_object()
+        # This locks a row in the dandiset table until the function is finished
+        # This is likely fine here since this function doesn't make any network calls,
+        # or otherwise have any long-running operations.
+        dandiset: Dandiset = Dandiset.objects.select_for_update().get(pk=dandiset__pk)
```

Comment:
I think this code comment can be classified as a "why, not what" comment. i.e. it is explaining something that should be obvious from reading the code. 
```suggestion
        dandiset: Dandiset = Dandiset.objects.select_for_update().get(pk=dandiset__pk)
```

Reply:
Does this work since select for update needs to be done inside a transaction?

Reply:
@danlamanna I'm not sure. I looked at the documentation for [select_for_update](https://docs.djangoproject.com/en/5.1/ref/models/querysets/#select-for-update) and in their example, the `select_for_update` is used just outside of a transaction (at least that what it looks like to me)

Reply:
The reason why it works in those docs is because querysets are evaluated lazily. In that example, `entries` is evaluated by the for loop, which is inside the transaction.atomic context manager so it works.
 
https://github.com/dandi/dandi-archive/pull/2288/files#diff-07074e43bf5319fe74c218e44c401e8152e75396cc6e478f6de0c751b1726fbdR524

So in your case, you will need a transaction as Dan suggested.